solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/batch/SolrSearchSink.java

```diff
@@ -0,0 +1,216 @@
+  "The input fields coming from the previous stage of the pipeline are mapped to Solr fields. User can also specify " +
+  "the mode of the Solr to connect to. For example, Single Node Solr or SolrCloud.")
+public class SolrSearchSink extends BatchSink<StructuredRecord, SolrInputField, SolrInputDocument> {
+
+  private static final String SINGLE_NODE_SOLR_MODE = "Single Node Solr";
+  private static final String SOLR_CLOUD_MODE = "SolrCloud";
+  private final SolrSearchSinkConfig config;
+  private String uniqueKey;
+  private Map<String, String> outputFieldMappings;
+  private SolrClient solrClient;
+
+  private File tempDir;
+
+  public SolrSearchSink(SolrSearchSinkConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void prepareRun(BatchSinkContext batchSinkContext) throws Exception {
+    // Setting FileOutputFormat.setOutputPath in order to resolve the 'Output directory not set' error.
+    tempDir = Files.createTempDir();
+    Job job = batchSinkContext.getHadoopJob();
+    FileOutputFormat.setOutputPath(job, new Path(tempDir.getAbsolutePath() + "/temp"));
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    validateSolrConnectionString();
+    verifySolrConfiguration();
+    if (inputSchema != null) {
+      validateIdField(inputSchema);
+      validateInputFieldsDataType(inputSchema);
+    }
+  }
+
+  @Override
+  public void initialize(BatchRuntimeContext context) throws Exception {
+    uniqueKey = config.getIdField();
+    outputFieldMappings = config.getOutputFieldMappings();
+    solrClient = getSolrConnection();
+  }
+
+  @Override
+  public void transform(StructuredRecord structuredRecord, Emitter<KeyValue<SolrInputField, SolrInputDocument>> emitter)
+    throws Exception {
+    String solrFieldName;
+    SolrInputDocument document;
+
+    validateIdField(structuredRecord.getSchema());
+    validateInputFieldsDataType(structuredRecord.getSchema());
+
+    if (structuredRecord.get(uniqueKey) == null) {
+      return;
+    }
+    document = new SolrInputDocument();
+    for (Schema.Field field : structuredRecord.getSchema().getFields()) {
+      solrFieldName = field.getName();
+      if (outputFieldMappings.containsKey(solrFieldName)) {
+        document.addField(outputFieldMappings.get(solrFieldName), structuredRecord.get(solrFieldName));
+      } else {
+        document.addField(solrFieldName, structuredRecord.get(solrFieldName));
+      }
+    }
+    solrClient.add(document);
+    solrClient.commit();
+    emitter.emit(new KeyValue<>(document.getField(uniqueKey), document));
+  }
+
+  @Override
+  public void destroy() {
+    solrClient.shutdown();
+  }
+
+  @Override
+  public void onRunFinish(boolean succeeded, BatchSinkContext context) {
+    File directory = new File(tempDir.getAbsolutePath() + "/temp");
```

Comment:
this deletion can be removed if you use:
https://hadoop.apache.org/docs/r2.4.1/api/org/apache/hadoop/mapreduce/lib/output/NullOutputFormat.html